Kernel/Arch/init.cpp

```diff
@@ -420,7 +420,13 @@ void init_stage2(void*)
     for (auto* init_function = driver_init_table_start; init_function != driver_init_table_end; init_function++)
         (*init_function)();
 
-    StorageManagement::the().initialize(kernel_command_line().root_device(), kernel_command_line().is_force_pio(), kernel_command_line().is_nvme_polling_enabled());
+    StorageManagement::the().initialize(kernel_command_line().is_force_pio(), kernel_command_line().is_nvme_polling_enabled());
+    for (int i = 0; i < 5; ++i) {
+        if (StorageManagement::the().determine_boot_device(kernel_command_line().root_device()))
+            break;
+        dmesgln("Boot device {} not found waiting 2 seconds", kernel_command_line().root_device());
```

Comment:
Please either change this `dmesgln` to `dbgln` and put it behind a debug flag (with more reasonable text, like "Boot device {} was not found, sleeping for 2 seconds")...

Reply:
Not sure if this is the right debug flag, but better than nothing